eosio.system/src/eosio.system.cpp

```diff
@@ -217,12 +218,27 @@ namespace eosiosystem {
       set_resource_limits( newact, 0, 0, 0 );
    }
 
+   void native::setabi( account_name acnt, const bytes& abi ) {
+      eosio::multi_index< N(abihash), abi_hash>  table(_self,_self);
```

Comment:
I'd prefer to see this included as a typedef in eosio.system.hpp like the other tables. Especially since it is ultimately meant to be used by other contracts (or at least other parts of code).

Also, the table definition needs to be added to the ABI.